queries/__init__.py

```diff
@@ -253,6 +253,546 @@ def convert(m):
+            if text[-1] == "ein":
+                text.append(isl_num)
+            else:
+                text.append(isl_num + "ir")
+
+    if THOUSAND <= n < MILLION:
+        thousands, n = divmod(n, THOUSAND)
+
+        if thousands > 1:
+            text.append(number_to_neutral(thousands))
+        elif thousands == 1:
+            text.append("eitt")
+        # Singular/Plural form of "þúsund" is the same
+        text.append("þúsund")
+
+    if 100 <= n < THOUSAND:
+        hundreds, n = divmod(n, 100)
+
+        if hundreds > 1:
+            text.append(number_to_neutral(hundreds))
+            # Note: don't need to fix singular here as e.g.
+            # 2100 gets interpreted as "tvö þúsund og eitt hundrað"
+            # instead of "tuttugu og eitt hundrað"
+            text.append("hundruð")
+
+        elif hundreds == 1:
+            text.append("eitt")
+            text.append("hundrað")
+
+    if 20 <= n < 100:
+        tens, digit = divmod(n, 10)
+        tens *= 10
+
+        text.append(_TENS_NEUTRAL[tens])
+        if digit != 0:
+            text.append("og")
+            text.append(_SUB_20_NEUTRAL[digit])
+        n = 0
+
+    if 0 < n < 20:
+        text.append(_SUB_20_NEUTRAL[n])
+        n = 0
+
+    if len(text) > 2 and text[-2] != "og":
+        # Fix sentences with missing "og"
+        if text[-1] in _SUB_20_NEUTRAL.values() or text[-1] in _TENS_NEUTRAL.values():
+            # "fimm þúsund tuttugu" -> "fimm þúsund og tuttugu"
+            # "eitt hundrað fjögur" -> "eitt hundrað og fjögur"
+            text.insert(-1, "og")
+        elif (
+            len(text) >= 3
+            and not re.search(r"(hundr|þúsund|jarð|jón)", text[-2])
+            and text[-3] != "og"
+        ):
+            # TODO: This if statement can probably be improved
+
+            # If-statement catches errors like "eitt og hundrað milljónir",
+            # but fixes numbers such as:
+            # "fimm þúsund tvö hundruð" -> "fimm þúsund og tvö hundruð"
+            # "sextíu milljónir fjögur hundruð" -> "sextíu milljónir og fjögur hundruð"
+            text.insert(-2, "og")
+
+    # Fix e.g. "milljónir milljarðar" -> "milljónir milljarða"
+    number_string: str = re.sub(
+        r"(\S*(jónir|jarð[au]r?)) (\S*(jarð|jón))[ia]r", r"\1 \3a", " ".join(text)
+    )
+
+    return number_string
+
+
+def number_to_text(n: int, gender="hk") -> str:
+    """
+    Convert an integer into written Icelandic text in given gender (hk, kk, kvk).
+    Example:
+        302 -> "þrjú hundruð og tvær" (gender="kvk")
+        501 -> "fimm hundruð og einn" (gender="kk")
+    """
+    num_str = number_to_neutral(n)
+
+    if gender =="hk":
+        return num_str
+
+    nums = num_str.split()
+
+    if gender=="kk" and nums[-1] in _NUM_NEUT_TO_KK:
+        nums[-1] = _NUM_NEUT_TO_KK[nums[-1]]
+
+    elif gender=="kvk" and nums[-1] in _NUM_NEUT_TO_KVK:
+        nums[-1] = _NUM_NEUT_TO_KVK[nums[-1]]
+
+    return " ".join(nums)
```

Comment:
Nice solution!

Reply:
Needs black formatting, canonical to have spaces between equality comparison operator (==).

Reply:
Thank you!
Formatting fixed